### PR TITLE
Added authentication configuration for the api implementation

### PR DESCRIPTION
**Pull request by nicozar95**
## Summary
* Added authentication configuration for the api implementation

## Trello
https://trello.com/c/xVCvpxxM/26-implementar-autenticaci%C3%B3n

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module Api
+  module V1
+    class UsersController < AuthenticationController
+      def create
+        @user = User.create(user_params)
+        render json: '', status: :ok
```
Comment (juanisierra) [inline app/controllers/api/v1/users_controller.rb]: If the user can't be created it should return and error. Add the error cases please

File: db/migrate/20171110172056_devise_create_users.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ def change
       t.string :encrypted_password, null: false, default: ""
       t.string :first_name, null: false
       t.string :last_name, null: false
+      t.string :locale, null: false
```
Comment (juanisierra) [inline db/migrate/20171110172056_devise_create_users.rb:9]: Remember that you shouldn't edit migrations, you should create  a new one that edits the schema

File: db/migrate/XXXX_add_verification_code_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddVerificationCodeToUsers < ActiveRecord::Migration[5.0]
```
Comment (juanisierra) [inline db/migrate/XXXX_add_verification_code_to_users.rb]: Please rename this file and change XXXX for a valid timestamp

**Review by juanisierra — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/users_controller.rb [hunk 4]
```diff
@@ -2,8 +2,12 @@ module Api
   module V1
     class UsersController < AuthenticationController
       def create
-        @user = User.create(user_params)
-        render json: '', status: :ok
+        @user = User.new(user_params)
+        if @user.save
+          render json: @user
+        else
+          render json: { errors: @user.errors.full_messages }
```
Comment (juanisierra) [inline app/controllers/api/v1/users_controller.rb]: Please add a Bad Request status when returning

**Review by juanisierra — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 5]
```diff
@@ -29,6 +29,10 @@ gem 'jbuilder', '~> 2.5'
 gem 'slim'
 
 gem 'devise'
+
+gem 'versionist'
```
Comment (juanisierra) [inline Gemfile:33]: What do you need this gem for?

**Review by juanisierra — APPROVED** [review]
Check that gem and merge

File: Gemfile (hunk 5 above)
Comment (nicozar95) [reply Gemfile:33]: It is for testing

**Pull request merged**